lang/funcs/core/coresys/cpucount_fact.go

```diff
@@ -0,0 +1,201 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package coresys
+
+import (
+	"io/ioutil"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/purpleidea/mgmt/lang/funcs/facts"
+	"github.com/purpleidea/mgmt/lang/types"
+
+	"github.com/purpleidea/mgmt/lib/socketset"
+	"golang.org/x/sys/unix"
+
+	errwrap "github.com/pkg/errors"
+)
+
+const (
+	rtmGrps         = 0x1 // make me a multicast reciever
+	socketFile      = "pipe.sock"
+	cpuDevpathRegex = "/devices/system/cpu/cpu[0-9]"
+)
+
+// CPUCountFact is a fact that returns the current CPU count
+type CPUCountFact struct {
+	init      *facts.Init
+	closeChan chan struct{}
+}
+
+func init() {
```

Comment:
This should be higher up at the top like in the other files. It's so that if a programmer has a quick look, they won't be surprised by code running by itself.